### PR TITLE
Task bot 2417 implement rail section

### PR DESCRIPTION
**Pull request by kun2985**
# Description

This PR implement rail section

## Issue (optional)

Issue link: bot-2417

## Test
upgrade the DB
`honcho run ./manage.py db upgrade`

update and build dedicated proto
```
git submodule update
./setup.py build_pbf
```

then try to CRUD impacts with one or several 'rail_section' composite objects.
cf swagger documentation : https://petstore.swagger.io/?url=https://raw.githubusercontent.com/CanalTP/Chaos/feature-rail-section-bot-2416/documentation/swagger.yml

### CODE REVIEW

File: migrations/versions/5bf2d2bc5cb_add_rail_section.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    sa.PrimaryKeyConstraint('id')
+    )
+    op.execute("COMMIT")  # See https://bitbucket.org/zzzeek/alembic/issue/123
+    op.execute("ALTER TYPE pt_object_type ADD VALUE 'rail_section'")
+
+
+def downgrade():
+    op.drop_table('rail_section')
+    op.execute("ALTER TABLE pt_object ALTER COLUMN type TYPE text")
+    op.execute("DELETE FROM pt_object WHERE  type='rail_section'")
```
Comment (pthegner) [inline migrations/versions/5bf2d2bc5cb_add_rail_section.py:41]:
```suggestion
    op.execute("DELETE FROM pt_object WHERE type='rail_section'")
```

File: chaos/db_helper.py [hunk 2]
```diff
@@ -171,6 +171,61 @@ def fill_and_add_line_section(navitia, all_objects, pt_object_json):
     ptobject.insert_line_section(line_section)
     return ptobject
 
+def fill_and_add_rail_section(navitia, all_objects, pt_object_json):
+    """
+    :param navitia: Class Navitia
+    :param all_objects: dictionary of objects to be added in this session
+    :param pt_object_json: Flux which contains json information of pt_object
```
Comment (Fahmus) [inline chaos/db_helper.py:178]:
```suggestion
    :param pt_object_json: Flux which contains json information of ordered pt_object
```

File: chaos/db_helper.py [hunk 3]
```diff
@@ -297,6 +352,13 @@ def create_or_update_impact(disruption, json_impact, navitia, impact_id=None):
                     exceptions.InvalidJson = xxx_todo_changeme
                     raise
                 impact_bd.objects.append(ptobject)
+            if pt_object_json["type"] == 'rail_section':
+                try:
+                    ptobject = fill_and_add_rail_section(navitia, all_objects, pt_object_json)
+                except exceptions.ObjectUnknown as xxx_todo_changeme:
```
Comment (tchevily) [inline chaos/db_helper.py:358]: Wich kind of changes are you waiting with "xxx_todo_changeme"

File: chaos/db_helper.py [hunk 4]
```diff
@@ -171,6 +171,61 @@ def fill_and_add_line_section(navitia, all_objects, pt_object_json):
     ptobject.insert_line_section(line_section)
     return ptobject
 
+def fill_and_add_rail_section(navitia, all_objects, pt_object_json):
```
Comment (tchevily) [inline chaos/db_helper.py:174]: This function is too long. You should refactor it.

File: tests/populate_pb_test.py [hunk 5]
```diff
@@ -420,85 +420,6 @@ def test_disruption_raw():
     eq_(disruption_pb.impacts[2].HasField('notification_date'), False)
 
 
-def test_disruption_without_routes():
```
Comment (tchevily) [inline tests/populate_pb_test.py:423]: Why was this test deleted ?

**Review by tchevily — CHANGES_REQUESTED** [review]

**Review by tchevily — CHANGES_REQUESTED** [review]
Functional tests are missing

File: tests/populate_pb_test.py (hunk 5 above)
Comment (pthegner) [reply tests/populate_pb_test.py:423]: next test seems to be a duplicate of this one.

File: chaos/db_helper.py [hunk 6]
```diff
@@ -16,7 +17,10 @@ def fill_and_get_pt_object(navitia, all_objects, json, add_to_db=True):
         return all_objects[json["id"]]
 
     if not navitia.get_pt_object(json['id'], json['type']):
-        raise exceptions.ObjectUnknown()
+        raise exceptions.ObjectUnknown(
+            '{} {} doesn\'t exist'.format(
+                json['type'],
+                json['id']))
```
Comment (pthegner) [inline chaos/db_helper.py:23]:
```suggestion
            "{} '{}' doesn't exist".format(
                json['type'],
                json['id']))
```

**Review by tchevily — APPROVED** [review]

**Review by Fahmus — APPROVED** [review]

**Review by Msaglier — APPROVED** [review]

**Pull request merged**